tests/rust_reflection_test/src/lib.rs

```diff
@@ -1499,10 +1499,902 @@ fn test_verify_buffer_limit_max_size_fails() {
         ..Default::default()
     };
 
-    let res = verify_with_options(&buffer, &schema, &verify_options);
+    let safe_buffer = SafeBuffer::new_with_custom_options(&buffer, &schema, &verify_options);
 
-    assert!(res.is_err());
-    assert!(format!("{:#?}", res.err().unwrap()).contains("ApparentSizeTooLarge"));
+    assert!(safe_buffer.is_err());
+    assert!(format!("{:#?}", safe_buffer.err().unwrap()).contains("ApparentSizeTooLarge"));
+}
+
+#[test]
+fn test_safe_buffer_integer_same_type_succeeds() {
+    let buffer = create_test_buffer();
+    let schema_buffer = load_file_as_buffer("../monster_test.bfbs");
+    let schema = root_as_schema(schema_buffer.as_slice()).unwrap();
+    let safe_buffer = SafeBuffer::new(&buffer, &schema).unwrap();
+    let root_table = safe_buffer.get_root();
+
+    let value = safe_buffer.get_field_integer::<i16>(&root_table, "hp");
+
+    assert!(value.is_ok());
+    assert_eq!(value.unwrap(), Some(32767));
+}
+
+#[test]
+fn test_safe_buffer_integer_invalid_table_fails() {
+    let buffer = create_test_buffer();
+    let schema_buffer = load_file_as_buffer("../monster_test.bfbs");
+    let schema = root_as_schema(schema_buffer.as_slice()).unwrap();
+    let safe_buffer = SafeBuffer::new(&buffer, &schema).unwrap();
+    let another_buffer = vec![1];
+    let invalid_table = unsafe { Table::new(&another_buffer, 0) };
+
+    let value = safe_buffer.get_field_integer::<i16>(&invalid_table, "hp");
+
+    assert!(value.is_err());
+    assert_eq!(
+        value.unwrap_err().to_string(),
+        "Table or Struct doesn't belong to the buffer"
```

Comment:
Nit: we're using `thiserror` so we could look for a specific variant of the error enum, to not have to worry about the string wording changing:
```
assert_eq!(matches!(value.unwrap_err(), FlatbufferError::InvalidTableOrStruct))
```